solutions/devsprint-pedro-menezes-1/DeliveryAppChallenge/Modules/Product/Scenes/PedroTres/PTResetPasswordViewController.swift

```diff
@@ -126,32 +127,27 @@ extension PTResetPasswordViewController {
         createAccountButton.layer.borderColor = UIColor.blue.cgColor
         createAccountButton.setTitleColor(.blue, for: .normal)
         createAccountButton.backgroundColor = .white
-        
-        emailTextfield.setDefaultColor()
-        
-        if !email.isEmpty {
-            emailTextfield.text = email
-            emailTextfield.isEnabled = false
-        }
-        validateButton()
-    }
-    
-    //email
-    @IBAction func emailBeginEditing(_ sender: Any) {
-        emailTextfield.setEditingColor()
     }
+}
+
+extension PTResetPasswordViewController {
     
-    @IBAction func emailEditing(_ sender: Any) {
-        emailTextfield.setEditingColor()
-        validateButton()
+    func validateForm() -> Bool {
```

Comment:
essa funçao nao vai ser necessária de exitir
pq  ela esta fazendo duas coisas, ela esta validando o form e tb mudando o layout se nao for válido
como vc criou a funçao `isInValidEmail`, então aonde vc chama a validateForm, vc chamara a `isInValidEmail`
e caso seja false, lá nessa outra funçao vc chama uma funçao que só configura layout para cenário de erro
e caso seja true, vc chama outra função que vai começar o processo para bater na API